auth/src/include/firebase/auth/credential.h

```diff
@@ -122,6 +122,31 @@ class Credential {
   /// @endcond
 };
 
+/// Wraps phone number and verification information for authentication purposes.
+class PhoneAuthCredential : public Credential {
+ public:
+  // Default constructor.
+  PhoneAuthCredential();
+
+  /// Copy constructor.
+  PhoneAuthCredential(const PhoneAuthCredential& rhs);
+
+  /// Copy a Credential.
+  PhoneAuthCredential& operator=(const PhoneAuthCredential& rhs);
+
+  /// Gets the automatically retrieved SMS verification code if applicable.
+  /// This method is only supported on Android.
+  std::string sms_code() const;
```

Comment:
After looking into this a bit deeper, I think you only need to implement `std::string sms_code() const;` per platform. All the other implementation are pretty much the same if you leverage the parent constructor and copy operator.

Perhaps introduce a `credential_common.cc` and put all the constructors and copy operator there?

Reply:
Sure! I added a `credential.cc` file with the other common files `auth.cc` and `user.cc`. I put the constructors in there but the assignment operator is still platform dependent.